src/ChanperfTracker.ts

```diff
@@ -0,0 +1,145 @@
+import { EventEmitter } from 'events';
+
+export class ChanperfTracker {
+    constructor() {
+        this.emitter = new EventEmitter();
+        this.filterOutLogs = true;
+        this.chanperfHistory = [];
+    }
+
+    private emitter: EventEmitter;
+    private filterOutLogs: boolean;
+    private chanperfHistory: Array<ChanperfEventData>;
+
+    public on(eventname: 'chanperf-event', handler: (output: ChanperfEventData) => void);
+    public on(eventName: string, handler: (payload: any) => void) {
+        this.emitter.on(eventName, handler);
+        return () => {
+            if (this.emitter !== undefined) {
+                this.emitter.removeListener(eventName, handler);
+            }
+        };
+    }
+
+    private emit(eventName: 'chanperf-event', data: ChanperfEventData) {
+        this.emitter.emit(eventName, data);
+    }
+
+    public get getChanperfHistory(): Array<ChanperfEventData> {
+        return this.chanperfHistory;
+    }
+
+    /**
+     * Used to set wether the chanperf should be filtered from the console output
+     * @param outputLevel the consoleOutput from the launch config
+     */
+    public setConsoleOutput(outputLevel: string) {
+        this.filterOutLogs = (outputLevel !== 'full');
+    }
+
+    /**
+     * Clears the current chanperf history
+     */
+    public clearChanperfHistory() {
+        this.chanperfHistory = [];
+        this.emit('chanperf-event', this.chanperfHistory[0]);
+    }
+
+    /**
+     * Takes the debug output from the device and parses it for any chanperf information.
+     * Also if consoleOutput was not set to 'full' then any chanperf output will be filtered from the output.
+     * @param logLine
+     * @returns The debug output after parsing
+     */
+    public processLogLine(logLine: string): string {
```

Comment:
This function handles multi-line log output, so it shouldn't have the word `line` in it. Perhaps `processLogOutput` or even just `processLog`.

Reply:
Yea, I'll update this. I'll also update the same function name in the ￼RendezvousTracker.